test/interval.jl

```diff
@@ -1,10 +1,20 @@
+Base.isinf(x::Char) = false
+Base.isinf(x::TimeType) = false
```

Comment:
piracy?

Reply:
could there at least be a comment?

Reply:
Definitely, it's just for tests though so it's not a big deal

Reply:
should we just define and use our own `_isinf`?

Reply:
I may open a PR to get `isinf(::TimeType)` implemented but I very much doubt I would get a PR for `isinf(::Char)` accepted

Reply:
If this really bothers people I can do:
```julia
_isinf(x) = isinf(x)
_isinf(::Char) = false
_isinf(::TimeType) = false
```

Reply:
i'd slightly prefer that as it make the code easier to follow (to my eyes) as it make clearly we've added some particulars for our use-case

Reply:
How about another compromise?
```julia
isinf(x) = Base.isinf(x)
isinf(::Char) = false
isinf(::TimeType) = false
```

Reply:
Something similar: https://github.com/invenia/Intervals.jl/pull/104/files#r444362741. We could actually have a generic `isinf` implemented as:
```julia
isinf(x) = !isfinite(x) && !isnan(x)
isfinite(x) = iszero(x - x)
isnan(x) = x != x
```
The only issue I see with this is that the generic `isnan` may not always hold for all types